### DISCUSSION TITLE
Labeling Volume Mounts via REST API

### DISCUSSION THREAD

**Discussion by quarckster** [containers/podman#12095] [Q&A]
We can label volume mounts by providing one of two suffixes `:z` or `:Z`, e.g. `podman run -it -v some_path:some_path:z some_image`. How can I label volumes via REST API?

**Comment by mheon** [comment]
Do you want to use the Libpod or Compat API endpoints?

**Reply by quarckster** [reply]
I create containers using this endpoint https://docs.podman.io/en/latest/_static/api.html#operation/ContainerCreateLibpod with `mount` attribute.

**Reply by mheon** [reply]
@jwhonce @baude The docs there look wrong. The Libpod endpoints accept a Specgen, but that's definitely not a Specgen being documented there.

**Comment by rhatdan** [comment]
I take it the :z and :Z do not work via the API?

**Reply by rhatdan** [reply]
Adding them to the destination path, I mean?

**Reply by quarckster** [reply]
I will give it a try, thanks @rhatdan

**Reply by mheon** [reply]
Adding them to the path won't work.

There is a field in SpecGen for volume options which should be used (https://github.com/containers/podman/blob/main/pkg/specgen/specgen.go#L240 points to https://github.com/opencontainers/runtime-spec/blob/master/specs-go/config.go#L110-L120 which contains an `Options` array - it should go there). The problem is that our ContainerCreate documentation does not reflect this situation - it seems to be using the Docker-compat structs?
